### ISSUE TITLE
failed to authenticate when downloading repository

### ISSUE THREAD

**Issue by adamdecaf** [databricks/click#38]
This is probably a `cargo` issue, but I'm not that familiar with rust.

I was trying `cargo install` and ran into the following: 

```
$ cargo install -vv
  Installing click v0.3.1 (file:///Users/adam/code/src/github.com/databricks/click)
    Updating registry `https://github.com/rust-lang/crates.io-index`
error: failed to compile `click v0.3.1 (file:///Users/adam/code/src/github.com/databricks/click)`, intermediate artifacts can be found at `/Users/adam/code/src/github.com/databricks/click/target`

Caused by:
  failed to fetch `https://github.com/rust-lang/crates.io-index`

Caused by:
  failed to authenticate when downloading repository
attempted ssh-agent authentication, but none of the usernames `git` succeeded

Caused by:
  error authenticating: ; class=Ssh (23)
```

I have the following in my `~/.gitconfig` for Go projects, but commenting this out let me run `cargo install` just fine.

```
[url "git@github.com:"]
	insteadOf = https://github.com/
```

**Comment by adamdecaf** [comment]
This is a cargo issue. See https://github.com/rust-lang/cargo/issues/2078 There's another fix (https://github.com/rust-lang/cargo/issues/1851#issuecomment-288791836) offered using `ssh-agent` on OSX.

**Comment by Olshansk** [comment]
> This is a cargo issue. See [rust-lang/cargo#2078](https://github.com/rust-lang/cargo/issues/2078) There's another fix ([rust-lang/cargo#1851 (comment)](https://github.com/rust-lang/cargo/issues/1851#issuecomment-288791836)) offered using `ssh-agent` on OSX.

Confirmed that this still works in 2023
